nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,75 @@
+const express = require("express");
+const router = express.Router();
+
+const meals = require("../data/meals.json");
+
+router.get("/", async (request, response) => {
+  try {
+    let newMeals = meals;
+    const { maxPrice, searchTitle, createdAfter, limit } = request.query;
+    const queryArray = [
+      "obj",
+      "maxPrice",
+      "searchTitle",
+      "createdAfter",
+      "limit",
+    ];
+    const userQueries = Object.keys(request.query);
+    const comparedQuries = [];
+
+    for (let i = 0; i < queryArray.length; i++) {
+      for (let j = 0; j < userQueries.length; j++) {
+        if (userQueries[j] === queryArray[i]) {
+          comparedQuries.push(userQueries[j]);
+        }
+      }
+    }
```

Comment:
This whole block can be replaced by the "intersection" function https://underscorejs.org/#intersection

Reply:
I thought using nested for loop seems a bit silly so I made a new one which is "meals-router2"(FYI)

and when it comes to change to the intersection,  with this nested for loop, can it be something like this?

`comparedQuries.intersection(queryArray, userQueries);`